spec/listformat.html

```diff
@@ -4,149 +4,108 @@ <h1>ListFormat Objects</h1>
+          1. Let _subst_ be _placeables_.[[&lt;_part_&gt;]].
+          1. If Type(_subst_) is List, then
+            1. For each element _s_ of _subst_ in List order, do
+              1. Append _s_ as the last element of _result_.
           1. Else,
-            1. If ? HasOwnProperty(_placeables_, _part_), then
-              1. Throw a TypeError exception..
-            1. Set _subst_ to _placeables_.[[_part_]].
-            1. If Type(_subst_) is a List, then
-              1. Let _len_ be ? ToLength(? Get(_subst_, `"length"`)).
-              1. Set _i_ to *0*.
-              1. Repeat, while _i_ &lt; _len_
-                1. Add _subst_.[_i_] as a new element on the list _result_.
-            1. Else,
-              1. Add _subst_ as a new element on the list _result_.
+            1. Append _subst_ as the last element of _result_.
+          1. Set _nextIndex_ to _endIndex_ + 1.
+          1. Set _beginIndex_ to ! Call(%StringProto_indexOf%, _pattern_, &laquo; `"{"`, _nextIndex_ &raquo;).
+        1. If _nextIndex_ is less than _length_, then
+          1. Let _literal_ be the substring of _pattern_ from position _nextIndex_, inclusive, to position _length_, exclusive.
+          1. Append a new Record { [[Type]]: `"literal"`, [[Value]]: _literal_ } as the last element of _result_.
         1. Return _result_.
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-createpartsfromlist" aoid="CreatePartsFromList">
-      <h1>CreatePartsFromList (listFormat, list)</h1>
+      <h1>CreatePartsFromList (_listFormat_, _list_)</h1>
 
       <p>
-        The *CreatePartsFromList* abstract operation is called with arguments
-        _listFormat_ (which must be an object initialized as a ListFormat) and _list_ (which must be a List), and creates the corresponding list of parts according to the effective locale and the formatting options of _listFormat_.
-        Each part is a Record with two fields: Type, which must be a string with values "element" or "literal", and Value which must be a string or a number.
+        The CreatePartsFromList abstract operation is called with arguments _listFormat_ (which must be an object initialized as a ListFormat) and _list_ (which must be a List of String values), and creates the corresponding list of parts according to the effective locale and the formatting options of _listFormat_.
+        Each part is a Record with two fields: [[Type]], which must be a string with values "element" or "literal", and [[Value]] which must be a string or a number.
         The following steps are taken:
       </p>
 
       <emu-alg>
-        1. Let _len_ be the length of _list_.
-        1. If _list_ contains any element _value_ such that Type(_value_) is not String, throw a *TypeError* exception.
-        1. If _len_ is *0*, then
+        1. Let _size_ be the number of elements of _list_.
+        1. If _size_ is 0, then
           1. Return a new empty List.
-        1. If _len_ is *1*, then
```

Comment:
It's true that removing the special case is correct, but subjectively, I found the algorithm to be clearer when the special case was included. (I'm fine either way though; no need to revert).